File: Q0024_Swap_Nodes_in_Pairs/Solution.java
package Q0024_Swap_Nodes_in_Pairs;

/**
 * @ Description:
 * @ Date: Created in 17:08 2019-04-06
 * @ Author: Anthony_Duan
 */
public class Solution {
    class ListNode {
        int val;
        ListNode next;

        ListNode(int x) {
            val = x;
        }
    }

    public ListNode swapPairs(ListNode head) {

        ListNode dummyHead = new ListNode(-1);
        dummyHead.next = head;
        ListNode p = dummyHead;
        ListNode node1, node2, next = null;

        while (p.next != null && p.next.next != null) {
            node1 = p.next;
            node2 = node1.next;
            next = node2.next;

            p.next = node2;
            node2.next = node1;
            node1.next = next;
            p = node1;
        }

        return dummyHead.next;


    }
}

File: Q0024_Swap_Nodes_in_Pairs/Solution.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution:
    def swapPairs(self, head: ListNode) -> ListNode:
        dummy_head = ListNode
        dummy_head.next = head
        p = dummy_head
        while p.next is not None and p.next.next is not None:
            node1, node2, next = p.next, p.next.next, p.next.next.next
            p.next = node2
            node2.next = node1
            node1.next = next
            p = node1
        return dummy_head.next
